packages/components/src/test/properties/inputs/UniquePropertyValuesSelector.test.tsx

```diff
@@ -478,6 +481,281 @@ describe("UniquePropertyValuesSelector", () => {
     });
   });
 
+  it("menu shows `No values` message when an error occurs loading values", async () => {
+    getDistinctValuesIteratorStub.rejects();
+
+    const { getByText, user } = render(
+      <TestComponentWithPortalTarget>
+        <UniquePropertyValuesSelector property={propertyDescription} onChange={() => {}} imodel={testImodel} descriptor={descriptor} />,
+      </TestComponentWithPortalTarget>,
+    );
+
+    // open menu
+    const menuSelector = await waitFor(() => getByText("unique-values-property-editor.select-values"));
+    await user.click(menuSelector);
+
+    // ensure only one page was loaded
+    await waitFor(() => getByText("unique-values-property-editor.no-values"));
+    expect(getDistinctValuesIteratorStub).to.be.calledOnce;
+  });
+
+  describe("search", () => {
+    function matchPageStart(start: number) {
+      return sinon.match((options: { paging: { start: number } }) => options.paging.start === start);
+    }
+
+    it("filters values based on search input", async () => {
+      getDistinctValuesIteratorStub.withArgs(matchPageStart(0)).resolves({
+        total: 2,
+        items: createAsyncIterator([
+          { displayValue: "Value1", groupedRawValues: ["TestValue1"] },
+          { displayValue: "TestValue2", groupedRawValues: ["TestValue2"] },
+        ]),
+      });
+
+      const { getByText, user, container } = render(
+        <TestComponentWithPortalTarget>
+          <UniquePropertyValuesSelector property={propertyDescription} onChange={() => {}} imodel={testImodel} descriptor={descriptor} />,
+        </TestComponentWithPortalTarget>,
+      );
+
+      // open menu
+      const menuSelector = await waitFor(() => getByText("unique-values-property-editor.select-values"));
+      await user.click(menuSelector);
+
+      // type in
+      const searchSelector = await waitFor(() => container.querySelector(".presentation-async-select-values-container"));
+      await user.type(searchSelector!, "test");
+
+      // ensure only the searched for value is shown
+      await waitFor(() => getByText("TestValue2"));
```

Comment:
I think `expect(queryByText("TestValue2")).to.not.be.null` would be more descriptive. Also we probably should check that `Value1` is not visible. Same in other tests.

Reply:
Updated the tests.